### PR TITLE
[Inductor][CPP] Add oneDNN BRGEMM config for Half cpp gemm template

### PR DESCRIPTION
**Pull request by CaoE**
`kernel_micro_gemm` generated using BRGEMM:
```
template <bool accum>
inline void kernel_micro_gemm(
    const half* __restrict__ A,
    const half* __restrict__ B,
    float* __restrict__ C,
    int64_t M,
    int64_t N,
    int64_t K,
    int64_t lda,
    int64_t ldb,
    int64_t ldc
) {
    at::native::cpublas::brgemm(
      M, N, K,
      lda, ldb, ldc,
      1.f, accum ? 1.f : 0.f,
      A,
      B,
      C);
}
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

### CODE REVIEW

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: torch/_inductor/codegen/cpp_micro_gemm.py [hunk 1]
```diff
@@ -741,6 +741,69 @@ def get_b_layout(self):
...
+
+    def codegen_finalize(
+        self,
+        kernel: CppTemplateKernel,
+    ) -> str:
+        return "at::native::cpublas::brgemm_release();"
+
+    def get_b_layout(self):
+        assert self.input_dtype == torch.half
+        return LayoutType.VNNI2
```
[the remark this answers is not in the document]
Comment (CaoE) [reply torch/_inductor/codegen/cpp_micro_gemm.py:804]: Yes. brgemm also supports non-AMX compute. Will enable it on non-AMX cases.

File: torch/csrc/cpu/Module.cpp [hunk 2]
```diff
@@ -13,6 +13,7 @@ void initModule(PyObject* module) {
   cpu.def("_is_avx512_vnni_supported", at::cpu::is_avx512_vnni_supported);
   cpu.def("_is_avx512_bf16_supported", at::cpu::is_avx512_bf16_supported);
   cpu.def("_is_amx_tile_supported", at::cpu::is_amx_tile_supported);
+  cpu.def("_is_amx_fp16_supported", at::cpu::is_amx_fp16_supported);
```
[the remark this answers is not in the document]
Comment (CaoE) [reply torch/csrc/cpu/Module.cpp:17]: Add `_is_amx_tile_supported` for `_is_amx_fp16_supported`.

File: torch/_inductor/codegen/cpp_micro_gemm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (CaoE) [reply torch/_inductor/codegen/cpp_micro_gemm.py]: Added non-AMX compute support.

**Review by jgong5 — CHANGES_REQUESTED** [review]

File: test/inductor/test_cpu_select_algorithm.py [hunk 3]
```diff
@@ -134,6 +134,14 @@ def _check_amx_counter(self, vec_amx):
         else:
             self.assertEqual(counters["inductor"]["cpp_micro_gemm_amx_counter"], 0)
 
+    def _check_brgemm_counter(self, vec_amx):
+        if vec_amx and (
+            torch.cpu._is_amx_fp16_supported() or torch.cpu._is_avx512_fp16_supported()
```
[the remark this answers is not in the document]
Comment (CaoE) [reply test/inductor/test_cpu_select_algorithm.py]: Checking avx512_fp16 is enough. Modified. Thanks.

[the remark this answers is not in the document]
Comment (jgong5) [reply test/inductor/test_cpu_select_algorithm.py]: No, my question is why we care about avx512_fp16 here. We only use AMX FP16, right?

**Review by jgong5 — COMMENTED** [review]
Please explain why we need avx512 fp16 here.

File: test/inductor/test_cpu_select_algorithm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (CaoE) [reply test/inductor/test_cpu_select_algorithm.py]: Removed non-amx fp16 support. Only check AMX FP16 here.

**Comment by CaoE** [comment]
> Please explain why we need avx512 fp16 here.

Removed non-amx fp16 support.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**